jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -991,6 +983,28 @@ private static final MethodHandle resolveInvokeDynamic(long j9class, String name
 		return result;
 	}
 	
+	/**
+	 * This helper calls MethodType.fromMethodDescriptorString(...) but throws NoClassDefFoundError
+	 * instead of TypeNotPresentException during the VM resolve stage.
+	 *
+	 * @param methodDescriptor - the method descriptor string
+	 * @param loader - the ClassLoader to be used or null for System ClassLoader
+	 *
+	 * @return a MethodType object representing the method descriptor string
+	 *
+	 * @throws IllegalArgumentException - if the string is not well-formed
+	 * @throws NoClassDefFoundError - if a named type cannot be found
+	 */
+	private static final MethodType vmResolveFromMethodDescriptorString(String methodDescriptor, ClassLoader loader) throws Throwable {
+		MethodType type = null;
```

Comment:
Initializing a variable to `null` here is not necessary.

Reply:
fixed.